awswrangler/mysql.py

```diff
@@ -340,16 +362,33 @@ def to_sql(
                 df.reset_index(level=df.index.names, inplace=True)
             column_placeholders: str = ", ".join(["%s"] * len(df.columns))
             insertion_columns = ""
+            upsert_columns = ""
+            upsert_str = ""
             if use_column_names:
                 insertion_columns = f"({', '.join(df.columns)})"
+            if mode.lower().strip() == "upsert_duplicate_key":
```

Comment:
Since the `lower().strip()` operation was already performed once, there is no need to run it here again